cmd/bnscli/cmd_gov.go

```diff
@@ -67,6 +59,46 @@ transaction (ie signatures) are being dropped.
 		option.Option = &app.ProposalOptions_ResetRevenueMsg{
 			ResetRevenueMsg: msg,
 		}
+	case *app.BatchMsg:
+		msgs, err := msg.MsgList()
+		if err != nil {
+			return fmt.Errorf("cannot extract messages: %s", err)
+		}
+		var messages []app.ProposalBatchMsg_Union
+		for _, m := range msgs {
+			switch m := m.(type) {
+			case *cash.SendMsg:
+				messages = append(messages, app.ProposalBatchMsg_Union{
+					Sum: &app.ProposalBatchMsg_Union_SendMsg{
+						SendMsg: m,
+					},
+				})
+			case *escrow.ReleaseEscrowMsg:
+				messages = append(messages, app.ProposalBatchMsg_Union{
+					Sum: &app.ProposalBatchMsg_Union_ReleaseEscrowMsg{
+						ReleaseEscrowMsg: m,
+					},
+				})
+			case *distribution.ResetRevenueMsg:
+				messages = append(messages, app.ProposalBatchMsg_Union{
+					Sum: &app.ProposalBatchMsg_Union_ResetRevenueMsg{
+						ResetRevenueMsg: m,
+					},
+				})
+			default:
+				return fmt.Errorf("message %T is not supported as batch proposal", m)
+			}
+
+		}
+		option.Option = &app.ProposalOptions_BatchMsg{
+			BatchMsg: &app.ProposalBatchMsg{
+				Messages: messages,
+			},
+		}
+	case *app.ProposalBatchMsg:
```

Comment:
I like this one... It is a noop for safety

Reply:
But... this will never be the case for a top-level bnsd/app.Tx.
It ca be removed.